abdm_integrator/user_auth/views.py

```diff
@@ -0,0 +1,134 @@
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_202_ACCEPTED
+from rest_framework.views import APIView
+
+from abdm_integrator.const import AuthenticationMode
+from abdm_integrator.exceptions import ABDMGatewayCallbackTimeout, ABDMGatewayError
+from abdm_integrator.settings import app_settings
+from abdm_integrator.user_auth.const import CALLBACK_RESPONSE_CACHE_TIMEOUT, UserAuthGatewayAPIPath
+from abdm_integrator.user_auth.exceptions import (
+    user_auth_error_response_handler,
+    user_auth_gateway_error_response_handler,
+)
+from abdm_integrator.user_auth.serializers import (
+    AuthConfirmSerializer,
+    AuthFetchModesSerializer,
+    AuthInitSerializer,
+    GatewayAuthOnConfirmSerializer,
+    GatewayAuthOnFetchModesSerializer,
+    GatewayAuthOnInitSerializer,
+)
+from abdm_integrator.utils import ABDMCache, ABDMRequestHelper, poll_and_pop_data_from_cache
+
+
+class UserAuthBaseView(APIView):
+    authentication_classes = [app_settings.AUTHENTICATION_CLASS]
+    permission_classes = [IsAuthenticated]
+
+    def get_exception_handler(self):
+        return user_auth_error_response_handler.get_exception_handler()
+
+    @staticmethod
+    def generate_response_from_callback(response_data):
+        if not response_data:
+            raise ABDMGatewayCallbackTimeout()
+        if response_data.get('error'):
+            error = response_data['error']
+            raise ABDMGatewayError(error.get('code'), error.get('message'))
+        return Response(status=HTTP_200_OK, data=response_data['auth'])
+
+
+class UserAuthGatewayBaseView(APIView):
+    serializer_class = None
+
+    def get_exception_handler(self):
+        return user_auth_gateway_error_response_handler.get_exception_handler()
+
+    def _post(self, request, format=None):
```

Comment:
Consider making this into a separate Mixin to use only where necessary since it's not getting used in all the classes that are inheriting from this.

Reply:
This is actually being used at all the three places.

Reply:
Okay, then why not also implement post itself here directly?

Reply:
Addressed in https://github.com/dimagi/abdm-python-integrator/pull/4/commits/ba1d2259d60116441c14273ad4bb16518283f463